src/convertToObject.js

```diff
@@ -12,7 +12,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const newArr = sourceString
+  .split(';')
+  .map(val => val.trim().split(':'))
+  .filter(val => val.length > 1);
+
+  const newObj = {};
+  newArr.map((arr) => (newObj[arr[0].trim()] = arr[1].trim()))
```

Comment:
method map is a clear function, so you shouldn't modify a data in this method
other way, you can use method forEach for that